File: src/list/get_max_sublist_sum.py
def get_max_sublist_sum(arr: list[int], length: int):
  max_count = 0
  max_count_start_index = 0
  current_count = 0
  for index, element in enumerate(arr):
    current_count += element
    if index < length:
      max_count = current_count
    else:
      current_count -= arr[index - length]
      if current_count > max_count:
        max_count = current_count
        max_count_start_index = index - length + 1
  return arr[max_count_start_index:max_count_start_index + length]

File: src/list/get_max_sublist_sum.ts
// ref: https://github.com/soygul/QuanticDev/blob/master/algorithms/dynamic-programming/sliding-window/max-subarray-sum.js

export default function getMaxSublistSum(input: number[], sublistLength: number) {
  let maxCount = 0;
  let maxCountStartIndex = 0;
  let currentCount = 0;
  for (let index = 0; index < input.length; index++) {
    currentCount += input[index];
    if (index < sublistLength) {
      maxCount = currentCount;
    } else {
      currentCount -= input[index - sublistLength];
      if (currentCount > maxCount) {
        maxCount = currentCount;
        maxCountStartIndex = index - sublistLength + 1;
      }
    }
  }
  return input.slice(maxCountStartIndex, maxCountStartIndex + sublistLength);
}
